src/Http/BaseApplication.php

```diff
@@ -304,6 +304,8 @@ public function services(ContainerInterface $container): void
     public function handle(
         ServerRequestInterface $request
     ): ResponseInterface {
+        $this->getContainer()->add(ServerRequest::class, $request);
```

Comment:
We could make a local variable for the container here to save a method call on line 310.

Reply:
Sure, I'll amend that.